lib/flag_shih_tzu.rb

```diff
@@ -242,118 +283,131 @@ def flag_keys(colmn = DEFAULT_COLUMN_NAME)
+      # Supposedly Rails 2.3 takes care of this, but this precaution is needed for backwards compatibility
+      has_table = has_ar ? connection.tables.include?(custom_table_name) : true
+      if has_table
+        found_column = columns.find { |column| column.name == colmn }
+        # If you have not yet run the migration that adds the 'flags' column
+        #   then we don't want to fail,
+        #   because we need to be able to run the migration
+        # If the column is there but is of the wrong type,
+        #   then we must fail, because flag_shih_tzu will not work
+        if found_column.nil?
+          warn(
+            %[Error: Column "#{colmn}" doesn't exist on table "#{custom_table_name}". Did you forget to run migrations?]
+          )
           return false
+        elsif found_column.type != :integer
+          raise IncorrectFlagColumnException.new(
+            %[Table "#{custom_table_name}" must have an integer column named "#{colmn}" in order to use FlagShihTzu.]
+          )
         end
-
-        true
+      else
+        # ActiveRecord gem may not have loaded yet?
+        warn(
+          %[FlagShihTzu#has_flags: Table "#{custom_table_name}" doesn't exist.  Have all migrations been run?]
+        ) if has_ar
+        return false
       end
 
-      def sql_condition_for_flag(flag, colmn, enabled = true, custom_table_name = self.table_name)
-        check_flag(flag, colmn)
+      true
+    end
 
-        if flag_options[colmn][:flag_query_mode] == :bit_operator
-          # use & bit operator directly in the SQL query.
-          # This has the drawback of not using an index on the flags colum.
-          "(#{custom_table_name}.#{colmn} & #{flag_mapping[colmn][flag]} = #{enabled ? flag_mapping[colmn][flag] : 0})"
-        elsif flag_options[colmn][:flag_query_mode] == :in_list
-          # use IN() operator in the SQL query.
-          # This has the drawback of becoming a big query when you have lots of flags.
-          neg = enabled ? "" : "not "
-          "(#{custom_table_name}.#{colmn} #{neg}in (#{sql_in_for_flag(flag, colmn).join(',')}))"
-        else
-          raise NoSuchFlagQueryModeException
-        end
+    def sql_condition_for_flag(flag, colmn, enabled = true, custom_table_name = table_name)
+      check_flag(flag, colmn)
+
+      if flag_options[colmn][:flag_query_mode] == :bit_operator
+        # use & bit operator directly in the SQL query.
+        # This has the drawback of not using an index on the flags colum.
+        %[(#{custom_table_name}.#{colmn} & #{flag_mapping[colmn][flag]} = #{enabled ? flag_mapping[colmn][flag] : 0})]
+      elsif flag_options[colmn][:flag_query_mode] == :in_list
+        # use IN() operator in the SQL query.
+        # This has the drawback of becoming a big query
+        #   when you have lots of flags.
+        neg = enabled ? "" : "not "
+        %[(#{custom_table_name}.#{colmn} #{neg}in (#{sql_in_for_flag(flag, colmn).join(",")}))]
```

Comment:
Line is too long. [95/80]<br>Prefer single-quoted strings inside interpolations.